src/CMakeLists.txt

```diff
@@ -799,9 +799,27 @@ add_definitions(-D__HEAP_SIZE=4096)
 #add_definitions(-DDEBUG)
 #add_definitions(-DDEBUG_NRF_USER)
 
-if (NOT CMAKE_BUILD_TYPE)
-    set(CMAKE_BUILD_TYPE "Release")
-endif ()
+# Debug configuration
+if (${CMAKE_BUILD_TYPE} STREQUAL "Debug")
+  add_definitions(-DDEBUG)
+  add_definitions(-DDEBUG_NRF_USER)
```

Comment:
The commented debug options can be removed, since it's controlled below.

Reply:
> The bin to hex change is blocking this currently, since there's concern that it may break compatibility with some companion apps. #256 also has this same change, so we'll reconsider that first.

The `objcopy` command is invoked by the CMake script as well, such that both the `bin` as well as the `hex` version are created. The binary file is exactly the same as before. Or are there concerns of tools depending on the inner working of the CMake compilation chain?